modules/swift/examples/code_snippets/SampleCodeTest.swift

```diff
@@ -1209,2824 +1208,1603 @@ class SampleCodeTest {
+        
+        // Return nil, if errors in key, certChain at this stage
+        guard let privateKey = privateKey, let certChain = certChain else {
             return nil
         }
-// end::p2p-tlsid-import-from-bundled[]
-
-// tag::p2p-tlsid-store-in-keychain[]
-// STORE THE IDENTITY AND ITS CERT CHAIN IN THE KEYCHAIN
-
+        // end::p2p-tlsid-import-from-bundled[]
+        
+        // tag::p2p-tlsid-store-in-keychain[]
+        // STORE THE IDENTITY AND ITS CERT CHAIN IN THE KEYCHAIN
+#if os(iOS)
+        // For iOS, need to save the identity into the KeyChain.
+        // Save or Update identity with a label so that it could be cleaned up easily
         // Store Private Key in Keychain
         let params: [String : Any] = [
             String(kSecClass):          kSecClassKey,
             String(kSecAttrKeyType):    kSecAttrKeyTypeRSA,
             String(kSecAttrKeyClass):   kSecAttrKeyClassPrivate,
-            String(kSecValueRef):       pKey
+            String(kSecValueRef):       privateKey
         ]
-        kcStatus = SecItemAdd(params as CFDictionary, nil)
-            if kcStatus != errSecSuccess {
-                print("Unable to store private key")
+        osStatus = SecItemAdd(params as CFDictionary, nil)
+        if osStatus != errSecSuccess {
+            print("Unable to store private key")
+            return nil
+        }
+        // Store all Certs for Id in Keychain:
+        var i = 0;
+        for cert in certChain {
+            let params: [String : Any] = [
+                String(kSecClass):      kSecClassCertificate,
+                String(kSecValueRef):   cert,
+                String(kSecAttrLabel):  "doco-sync-server"
+            ]
+            osStatus = SecItemAdd(params as CFDictionary, nil)
+            if osStatus != errSecSuccess {
+                print("Unable to store certs")
                 return nil
             }
-       // Store all Certs for Id in Keychain:
-       var i = 0;
-       for cert in thisCertChain! {
-            let params: [String : Any] = [
-                String(kSecClass):      kSecClassCertificate,
-                String(kSecValueRef):   cert,
-                String(kSecAttrLabel):  "doco-sync-server"
-                ]
-            kcStatus = SecItemAdd(params as CFDictionary, nil)
-                if kcStatus != errSecSuccess {
-                    print("Unable to store certs")
-                    return nil
-                }
-            i=i+1
+            i=i+1
+        }
+#else
+        let query: [String : Any] = [
+            String(kSecClass):          kSecClassCertificate,
+            String(kSecValueRef):       certs[0]
+        ]
+        
+        let update: [String: Any] = [
+            String(kSecClass):          kSecClassCertificate,
+            String(kSecValueRef):       certs[0],
+            String(kSecAttrLabel):      label
+        ]
+        
+        osStatus = SecItemUpdate(query as CFDictionary, update as CFDictionary)
```

Comment:
I don't understand the purpose of this code snippet that imports identity, extracts private key, add a new identity, and update? How are they related to CBL?

Reply:
I also don't know if it is the right thing to do to extract private key and add to a new item.

Reply:
@ibsoln could you check the relevance?

Reply:
@pasin @jayahariv It is part of a page we are not using at the moment. Leave for now at least.

Reply:
leaving it as it is.